### PR TITLE
PR : Add reporting for API

### PR DESCRIPTION
**Pull request by bitterpanda63**

### CODE REVIEW

File: aikido_firewall/background_process/api/http_api.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+            res = requests.post(
+                self.reporting_url + "api/runtime/events",
+                json=event,
+                timeout=timeout_in_sec,
+                headers=get_headers(token),
+            )
+        except requests.exceptions.ConnectionError:
+            return {"success": False, "error": "timeout"}
+        except Exception as e:
+            raise e
```
Comment (willem-delbare) [inline aikido_firewall/background_process/api/http_api.py]: debug log

File: aikido_firewall/background_process/heartbeats.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    if reporter.serverless:
+        logger.debug("Running in serverless environment, not starting heartbeats")
+        return
+    if not reporter.token:
+        logger.debug("No token provided, not starting heartbeats")
+        return
+
+    logger.debug("Starting heartbeats")
+
+    s.enter(0, 1, send_heartbeat_wrapper, (reporter, interval_in_secs, s))
```
Comment (willem-delbare) [inline aikido_firewall/background_process/heartbeats.py]: wat is s?

File: aikido_firewall/background_process/reporter.py [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+from aikido_firewall.helpers.token import Token
+from aikido_firewall import PKG_VERSION
+from aikido_firewall.background_process.heartbeats import send_heartbeats_every_x_secs
+
+
+class Reporter:
+    """Reporter class"""
+
+    timeout_in_sec = 5
+    heartbeat_secs = 10
```
Comment (willem-delbare) [inline aikido_firewall/background_process/reporter.py]: heartbeat is om de 10 min niet sec

File: aikido_firewall/background_process/api/http_api.py (hunk 1 above)
Comment (bitterpanda63) [reply aikido_firewall/background_process/api/http_api.py]: Done

File: aikido_firewall/sinks/pymysql.py [hunk 4]
```diff
@@ -39,8 +39,11 @@ def aikido_new_query(_self, sql, unbuffered=False):
 
         logger.info("sql_injection results : %s", json.dumps(result))
```
Comment (willem-delbare) [inline aikido_firewall/sinks/pymysql.py]: context_contains_for_sql_injection

File: aikido_firewall/sinks/pymysql.py [hunk 5]
```diff
@@ -39,8 +39,11 @@ def aikido_new_query(_self, sql, unbuffered=False):
...
         if result:
```
Comment (willem-delbare) [inline aikido_firewall/sinks/pymysql.py]: contains_inejction

Comment (bitterpanda63) [reply aikido_firewall/sinks/pymysql.py]: done

File: aikido_firewall/background_process/heartbeats.py (hunk 2 above)
Comment (bitterpanda63) [reply aikido_firewall/background_process/heartbeats.py]: Changed to event_scheduler

File: aikido_firewall/background_process/aikido_background_process.py [hunk 6]
```diff
@@ -6,11 +6,17 @@
...
 from queue import Queue
 from aikido_firewall.helpers.logging import logger
+from aikido_firewall.background_process.reporter import Reporter
+from aikido_firewall.helpers.should_block import should_block
+from aikido_firewall.helpers.token import get_token_from_env
+from aikido_firewall.background_process.api.http_api import ReportingApiHTTP
 
-REPORT_SEC_INTERVAL = 600  # 10 minutes
+
+REPORT_SEC_INTERVAL = 5  # 5 seconds
```
Comment (willem-delbare) [inline aikido_firewall/background_process/aikido_background_process.py]: moet 600 blijvne

File: aikido_firewall/background_process/reporter.py [hunk 7]
```diff
@@ -0,0 +1,172 @@
...
+from aikido_firewall.helpers.limit_length_metadata import limit_length_metadata
+from aikido_firewall.helpers.token import Token
+from aikido_firewall import PKG_VERSION
+from aikido_firewall.background_process.heartbeats import send_heartbeats_every_x_secs
+
+
+class Reporter:
+    """Reporter class"""
+
+    timeout_in_sec = 5
```
Comment (willem-delbare) [inline aikido_firewall/background_process/reporter.py]: timeout van wat precies?

File: aikido_firewall/background_process/aikido_background_process.py (hunk 6 above)
Comment (willem-delbare) [reply aikido_firewall/background_process/aikido_background_process.py]: buffer_attack_reports_time

File: aikido_firewall/background_process/aikido_background_process.py [hunk 8]
```diff
@@ -52,21 +59,36 @@ def __init__(self, address, key):
                     conn.close()
                     pid = os.getpid()
                     os.kill(pid, signal.SIGTERM)  # Kill this subprocess
+                elif data[0] == "READ_PROPERTY":
```
Comment (willem-delbare) [inline aikido_firewall/background_process/aikido_background_process.py]:
```suggestion
                elif data[0] == "READ_PROPERTY":  # meant to get config props
```

**Pull request merged**